File: server/models/pad.js
"use strict";

class Pad {

    constructor (id, name, padType, latitude, longitude, mapURL, retired, locationid, agencies, wikiURL, infoURLs) {
        this.id = id;
        this.name = name;
        this.padType = padType;
        this.latitude = latitude;
        this.longitude = longitude;
        this.mapURL = mapURL;
        this.retired = retired;
        this.locationid = locationid;
        this.agencies = agencies;
        this.wikiURL = wikiURL;
        this.infoURLs = infoURLs;
    }

    static fromJSON (json) {
        let x = JSON.parse(json);
        return new Pad(x.id, x.padType, x.latitude, x.longitude, x.mapURL, x.retired, x.locationid, x.agencies, x.wikiURL, x.infoURLs)
    }
}

File: src/main/java/com/eccff/launchbot/models/Pad.java
package com.eccff.launchbot.models;

import lombok.Builder;
import lombok.Data;

import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.Id;
import java.util.List;

/**
 * https://launchlibrary.net/1.2/pad
 */


@Data
@Entity
public class Pad {

    @Id
    private int id;
    private String name;
    private int padType;
    private String latitude;
    private String longitude;
    private String mapURL;
    private boolean retired;
    private int locationid;
    @ElementCollection(targetClass=Agency.class)
    private List<Agency> agencies;
    private String wikiURL;
    @ElementCollection(targetClass=String.class)
    private List<String> infoURLs;

}